### ISSUE TITLE
Error when using `property` and `cache` decorators with `Generic` class

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2911]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations

from functools import cache
from typing import Generic, TypeVar

T = TypeVar('T')


class Foo:
    @property
    @cache
    def prop(self, /) -> int:
        return 0

    def get_prop(self, /) -> int:
        return self.prop


class Bar(Generic[T]):
    @property
    @cache
    def prop(self, /) -> int:
        return 4

    def get_prop(self, /) -> int:
        return self.prop  # Error: Cannot access member "prop" for type "Bar[T@Bar]"


f = Foo()
f.prop
b: Bar[int] = Bar()
b.prop  # Error: Cannot access member "prop" for type "Bar[int]"
```

Two errors are generated as shown in the comments above.

**Expected behavior**
No errors should be generated.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.212

**Comment by erictraut** [comment]
It turns out that this has already been addressed by another bug fix that I checked in yesterday. So this will work in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.213, which I just published. It will also be included in the next release of pylance.
